app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -174,7 +180,45 @@ private void updateObstaclesCoordinates() {
             if (obstacle.getX() + obstacle.getWidth() < 0) {
                 it.remove();
                 score += 10;
+                if (activity.getSavedSignedInAccount() != null) {
+                    if (obstacle instanceof Cactus) {
+                        cactusScore += 1;
+                    } else if (obstacle instanceof Bat) {
+                        batScore += 1;
+                    } else if (obstacle instanceof Palm) {
+                        palmScore += 1;
+                    }
+                }
+            }
+        }
+    }
+
+    private void updateObstacleLeaderboards() {
+        if (activity.getSavedSignedInAccount() == null) {
+            return;
+        }
+        Resources resources = getResources();
+        for (int leaderboard : GameConstants.LEADERBOARD_LIST) {
+            String leaderboardId = resources.getString(leaderboard);
+            long score = ScoreManager.getScore(leaderboardId);
+            switch (leaderboard) {
+                case R.string.leaderboard_cactus_jumper:
+                    score += cactusScore;
+                    break;
+                case R.string.leaderboard_bat_avoider:
+                    score += batScore;
+                    break;
+                case R.string.leaderboard_palm_climber:
+                    score += palmScore;
+                    break;
+                case R.string.leaderboard_best_score:
+                case R.string.leaderboard_best_time:
+                    return;
+                default:
+                    Log.e(getClass().getName(), "An updateObstacleLeaderboards function: a leaderboard not found.");
```

Comment:
We don't usually have the name of the function in the error log. maybe we can have:
"Found un unknown obstacle" or something like this.